### PR TITLE
feat: Add unpublished filter

### PR DESCRIPTION
**Pull request by NarenderRajuB**

### CODE REVIEW

File: tests/test_admin_filters.py [hunk 1]
```diff
@@ -166,3 +168,51 @@ def test_site_filter(self):
             set(response_empty_site.context["cl"].queryset),
             set([])
         )
+
+
+@skipUnless(is_versioning_enabled(), 'Test only relevant for versioning')
+class UnpublishedFiltersTestCase(BaseAliasPluginTestCase):
+
+    def test_unpublished_filter(self):
```
Comment (adam-murray) [inline tests/test_admin_filters.py:176]: Could you make this test follow AAA? 

https://jamescooke.info/arrange-act-assert-pattern-for-python-developers.html

**Review by adam-murray — CHANGES_REQUESTED** [review]

**Review by adam-murray — APPROVED** [review]

File: tests/test_admin_filters.py [hunk 2]
```diff
@@ -166,3 +168,51 @@ def test_site_filter(self):
...
+        from djangocms_versioning.constants import UNPUBLISHED
+        from djangocms_versioning.models import Version
+
+        category = Category.objects.create(name='Alias Filter Category')
+
```
Comment (Aiky30) [inline tests/test_admin_filters.py:181]:
```suggestion
```

Redundant line for AAA

File: tests/test_admin_filters.py [hunk 3]
```diff
@@ -166,3 +168,51 @@ def test_site_filter(self):
...
+        unpublished_alias = AliasModel.objects.create(
+            category=category,
+            position=0)
+        expected_en_content = AliasContent.objects.create(
+            alias=alias,
+            name="EN Alias Content",
+            language="en"
+        )
+        Version.objects.create(content=expected_en_content, created_by=self.superuser)
+
```
Comment (Aiky30) [inline tests/test_admin_filters.py:195]:
```suggestion
```
Redundant line for AAA

File: tests/test_admin_filters.py [hunk 4]
```diff
@@ -166,3 +168,51 @@ def test_site_filter(self):
...
+            response_default = self.client.get(base_url)
+            # filter by unpublished hide
+            qs_default = response_default.context["cl"].queryset
+            response_unpublished = self.client.get(base_url + "?unpublished=1")
+            qs_unpublished = response_unpublished.context["cl"].queryset
+            # filter by unpublished show
+
+        # show all alias contents  excluding unpublished versions
+        self.assertEqual(set(qs_default), set([expected_en_content]))
+
```
Comment (Aiky30) [inline tests/test_admin_filters.py:216]:
```suggestion
```
Redundant line for AAA

File: djangocms_alias/filters.py [hunk 5]
```diff
@@ -76,3 +80,36 @@ def choices(self, changelist):
                 ),
                 "display": title,
             }
+
+
+if djangocms_versioning_enabled:
+    from djangocms_versioning.constants import UNPUBLISHED
+
+    class UnpublishedFilter(admin.SimpleListFilter):
+        title = _("unpublished")
```
Comment (Aiky30) [inline djangocms_alias/filters.py:89]:
```suggestion
        title = _("Unpublished")
```
To be consistent the first character is always upper cased

File: djangocms_alias/filters.py [hunk 6]
```diff
@@ -76,3 +80,36 @@ def choices(self, changelist):
...
+        parameter_name = "unpublished"
+
+        def lookups(self, request, model_admin):
+            return (("1", _("Show")),)
+
+        def queryset(self, request, queryset):
+            show = self.value()
+            if show == "1":
+                return queryset.filter(versions__state=UNPUBLISHED)
+            else:
```
Comment (Aiky30) [inline djangocms_alias/filters.py:99]: Redundant else clause. 
```suggestion
```

File: djangocms_alias/filters.py [hunk 7]
```diff
@@ -76,3 +80,36 @@ def choices(self, changelist):
...
+                return queryset.exclude(versions__state=UNPUBLISHED)
```
Comment (Aiky30) [inline djangocms_alias/filters.py:100]:
```suggestion
            return queryset.exclude(versions__state=UNPUBLISHED)
```

File: djangocms_alias/filters.py [hunk 8]
```diff
@@ -76,3 +80,36 @@ def choices(self, changelist):
...
+
+if djangocms_versioning_enabled:
+    from djangocms_versioning.constants import UNPUBLISHED
+
+    class UnpublishedFilter(admin.SimpleListFilter):
+        title = _("unpublished")
+        parameter_name = "unpublished"
+
+        def lookups(self, request, model_admin):
+            return (("1", _("Show")),)
```
Comment (Aiky30) [inline djangocms_alias/filters.py:93]: I'm not so sure about using 1, a reusable / future proofed alternative could have been to make this the version filter that only allows filtering by unpublished. No point wasting time changing this now, if more options are required in the future that person / request can rewrite the filter.

**Review by Aiky30 — CHANGES_REQUESTED** [review]
Minor formatting changes, and more importantly missing a changelog entry.

**Review by Bernardvdv — COMMENTED** [review]
Missing a changelog entry

File: tests/test_admin_filters.py (hunk 1 above)
Comment (NarenderRajuB) [reply tests/test_admin_filters.py:176]: Updated

**Review by adam-murray — APPROVED** [review]
All looks good to me 👍

**Review by Bernardvdv — APPROVED** [review]

**Review by Aiky30 — APPROVED** [review]

**Pull request merged**